app/src/main/java/edu/wpi/first/shuffleboard/app/plugin/PluginLoader.java

```diff
@@ -0,0 +1,189 @@
+package edu.wpi.first.shuffleboard.app.plugin;
+
+import edu.wpi.first.shuffleboard.api.data.DataTypes;
+import edu.wpi.first.shuffleboard.api.data.IncompatibleSourceException;
+import edu.wpi.first.shuffleboard.api.plugin.Plugin;
+import edu.wpi.first.shuffleboard.api.sources.DataSource;
+import edu.wpi.first.shuffleboard.api.sources.SourceTypes;
+import edu.wpi.first.shuffleboard.api.sources.recording.serialization.Serializers;
+import edu.wpi.first.shuffleboard.api.theme.Themes;
+import edu.wpi.first.shuffleboard.api.widget.Widget;
+import edu.wpi.first.shuffleboard.api.widget.Widgets;
+import edu.wpi.first.shuffleboard.app.sources.DestroyedSource;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Modifier;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.jar.JarFile;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+public class PluginLoader {
+
+  private static final Logger log = Logger.getLogger(PluginLoader.class.getName());
+
+  private static final PluginLoader defaultLoader = new PluginLoader();
+
+  private final Set<Plugin> loadedPlugins = new HashSet<>();
+  private final ObservableList<Plugin> knownPlugins = FXCollections.observableArrayList();
+
+  public static PluginLoader getDefault() {
+    return defaultLoader;
+  }
+
+  /**
+   * Loads a plugin jar and loads all plugin classes within.
+   *
+   * @param jarUri a URI representing  jar file to load plugins from
+   *
+   * @throws IOException if the jar file denoted by the URI could not be found or read
+   */
+  public void loadPluginJar(URI jarUri) throws IOException {
+    URL url = jarUri.toURL();
+    PrivilegedAction<URLClassLoader> getClassLoader = () -> {
+      return new URLClassLoader(new URL[]{url}, ClassLoader.getSystemClassLoader());
```

Comment:
I'd have to look into that. This is basically the same code that the original dashboard uses for loading jars at runtime and that has worked for the past 6+ years